fs/vfs/fs_sendfile.c

```diff
@@ -244,6 +244,11 @@ static ssize_t copyfile(FAR struct file *outfile, FAR struct file *infile,
 ssize_t file_sendfile(FAR struct file *outfile, FAR struct file *infile,
                       off_t *offset, size_t count)
 {
+  if (count == 0)
+    {
+      return 0;
```

Comment:
please add a nwarn() to let the user know that is tried to send a packet with 0 bytes. I think that kind message could have helped you with your application! ;-)

Reply:
Done.